lightning/src/util/ser.rs

```diff
@@ -463,6 +463,7 @@ macro_rules! impl_writeable_primitive {
 	}
 }
 
+impl_writeable_primitive!(u128, 16);
```

Comment:
As discussed offline I explored a number of approaches, e.g., utilizing a custom adapter in conjunction with handing through a `decode_custom_tlv` function. They seemed to be *almost* working on the decoding end (but don't really), and the encoding end is even trickier. Open for any suggestions how to move forward on this, otherwise I now broke the macro and now do custom de/ser as of https://github.com/lightningdevkit/rust-lightning/pull/1790/commits/3a7bd26d97a281f517dbf24f568de6df0e74fd68.